presto-hive/src/main/java/com/facebook/presto/hive/parquet/reader/ParquetReader.java

```diff
@@ -107,7 +126,42 @@ private boolean advanceToNextRowGroup()
         return true;
     }
 
-    public Block readBlock(ColumnDescriptor columnDescriptor, Type type)
+    public Block readStruct(Type type, List<String> path)
```

Comment:
`readStruct` modifies the `path` list passed to it. Makes it hard to track the state when reading the code, so why don't we make a copy here?